### ISSUE TITLE
cellFromPolygon(..., weights=TRUE) aggregation should ensure FLT4S datatype

### ISSUE THREAD

**Issue by johnbaums** [rspatial/raster#69]
If the user has set an integer data type as the default (e.g. `rasterOptions(datatype='INT1U')`), then `cellFromPolygons(r, p, weights=TRUE)` returns incorrect values.

For example:

```R
library(raster)
p <- readRDS(system.file('external/lux.rds', package='raster'))
r <- raster(extent(p), res=0.0001)

rasterOptions(progress='text', datatype='INT1U')
i1 <- cellFromPolygon(r, p[4,], weights=TRUE)[[1]]

rasterOptions(datatype='FLT4S')
i2 <- cellFromPolygon(r, p[4,], weights=TRUE)[[1]]

nrow(i1)
## [1] 954117
nrow(i2)
## [1] 956882

range(i1[, 'weight'])
## [1] 0.01 0.01
range(i2[, 'weight'])
## [1] 1e-04 1e-02
```

Since rasterization to the disaggregated raster will only ever result in 0s and 1s, it might be sensible to enforce `INT1U` (or in fact `LOG1S`) for the rasterization, thereby achieving a much smaller (in file size) disaggregated raster. 
https://github.com/rspatial/raster/blob/f564028d04f978f392b54d6e79ca0fcb2df70dee/R/rasterizePolygons.R#L139 

And enforce `FLT4S` for the aggregation, to ensure incompatible user-specified datatype default is ignored and fractional weights are permitted.
https://github.com/rspatial/raster/blob/f564028d04f978f392b54d6e79ca0fcb2df70dee/R/rasterizePolygons.R#L140

**Comment by rhijmans** [comment]
Thank you.  Fixed as suggested in in version 3.0-4

**Comment by johnbaums** [comment]
Fantastic, thanks Robert
